test/alternator/test_batch.py

```diff
@@ -402,4 +402,22 @@ def test_batch_get_item_partial(scylla_only, dynamodb, test_table_sn):
         responses.extend(reply['Responses'][test_table_sn.name])
     assert multiset(responses) == multiset(
         [{'p': p, 'c': i, 'content': content} for i in range(count)])
-    assert some_keys_were_unprocessed
\ No newline at end of file
+    assert some_keys_were_unprocessed
+
+# Test that if the batch read failure is total, i.e. all read requests
+# failed, it's reported as an error and not as a regular response with
+# UnprocessedKeys set to all given keys.
+def test_batch_get_item_full_failure(scylla_only, dynamodb, test_table_sn):
+    p = random_string()
+    content = random_string()
+    count = 10
+    with test_table_sn.batch_writer() as batch:
+        for i in range(count):
+            batch.put_item(Item={
+                'p': p, 'c': i, 'content': content})
+    responses = []
+    to_read = { test_table_sn.name: {'Keys': [{'p': p, 'c': c} for c in range(count)], 'ConsistentRead': True } }
+    # The error injection is permanent, so it will fire for each batch read.
+    scylla_inject_error(dynamodb, "alternator_batch_get_item", one_shot=False)
+    with pytest.raises(Exception, match="injection"):
```

Comment:
What type of exception do we get here? An internal server error? Maybe you can match one of DynamoDB's official error names and not just the word "injection"?
Also why Exception instead of ClientError?

Reply:
yeah, it's an internal server error, so I decided that the exact type is not very valuable here. I'll try to narrow the type down